.github/workflows/build.yml

```diff
@@ -1,35 +1,96 @@
-name: Test compilation
+name: CI
 
 on: [push, pull_request]
 
+env:
+  OCAML_VERSION: 4.07.1+flambda
+  CACHE_KEY: opam-flambda-noPslBase-${{github.base_ref}}-${{github.ref}} 
 jobs:
-  build:
+  build-deps:
     runs-on: ubuntu-latest
     steps:
+    - name: Try to restore build cache
+      id: cache
+      uses: actions/cache@v2
+      with:
+        path: "~/.opam"
+        key: ${{env.CACHE_KEY}}
+        restore-keys: |
+          opam-flambda-noPslBase--refs/heads/${{github.base_ref}}
 
-      - name: Try to restore opam cache
-        id: opam-cache
-        uses: actions/cache@v2
-        with:
-          path: "~/.opam"
-          key: opam-noPslBase-${{github.base_ref}}-${{github.ref}} 
-          restore-keys: |
-            opam-noPslBase--refs/heads/${{github.base_ref}}
-
-      - name: Install OCaml
-        uses: avsm/setup-ocaml@v1
-        with:
-          ocaml-version: 4.07.1
-
-      - name: Checkout code
-        uses: actions/checkout@v2
-        with:
-          fetch-depth: 1
-
-      - run: opam repo add coq-released https://coq.inria.fr/opam/released
-      - run: opam update
-      - run: opam upgrade
-      - run: opam install . --deps-only --with-doc --with-test
-      - run: opam list --installed
-
-      - run: opam exec -- make -j 2 all TIMED=1
+    - name: Install OCaml
+      if: steps.cache.outputs.cache-hit != 'true'
+      uses: avsm/setup-ocaml@v1
+      with:
+        ocaml-version: ${{env.OCAML_VERSION}}
+
+    - name: Checkout code
+      if: steps.cache.outputs.cache-hit != 'true'
```

Comment:
Do I understand correctly that the Coq installation now happens in the `build-deps` job, and the `build-full` and `build-fast` jobs use the cache stored by the `build-deps` job to compile (and that's why they fail if restoring fails)? I'm a bit surprised that GH actions don't allow sharing build files without caching.

Reply:
Yes, we use cache for that.